plugins/fluentd_telemetry_plugin/src/telemetry_parser.py

```diff
@@ -0,0 +1,417 @@
+from telemetry_constants import UFMTelemetryConstants
+from ufm_sdk_tools.src.xdr_utils import PortType,prepare_port_type_http_telemetry_filter
+from utils.logger import Logger, LOG_LEVELS
+from utils.utils import Utils
+
+class TelemetryParser:
+    """
+    UFM TelemetryParser class - to fetch and parse the telemetry data
+    """
+
+    PORT_CONSTANTS_KEYS = {
+        'timestamp': 'timestamp', 'source_id': 'source_id', 'tag': 'tag',
+        'node_guid': 'node_guid', 'port_guid': 'port_guid',
+        'sys_image_guid': 'sys_image_guid', 'aport': 'aport',
+        'port_num': 'port_num', 'node_description': 'node_description',
+        'm_label': 'm_label', 'port_label': 'port_label', 'status_message': 'status_message',
+        'Port_Number': 'Port_Number', 'Node_GUID': 'Node_GUID', 'Device_ID': 'Device_ID', 'device_id': 'Device_ID',
+        'mvcr_sensor_name': 'mvcr_sensor_name', 'mtmp_sensor_name': 'mtmp_sensor_name',
+        'switch_serial_number': 'switch_serial_number', 'switch_part_number': 'switch_part_number'
+        }
+    NORMAL_PORT_ID_KEYS = {'node_guid', 'Node_GUID', 'port_guid', 'port_num', 'Port_Number', 'Port'}
+    AGG_PORT_ID_KEYS = {'sys_image_guid', 'aport'}
+    PORT_TYPE_KEY = 'port_type'
+
+    def __init__(self, conf_parser, monitor_streaming_mgr, _last_streamed_data_sample_per_endpoint, attr_mngr):
+        self.config_parser = conf_parser
+        self.streaming_metrics_mgr = monitor_streaming_mgr
+        self.last_streamed_data_sample_per_endpoint = _last_streamed_data_sample_per_endpoint
+        self.meta_fields = self.config_parser.get_meta_fields()
+        self.attributes_mngr = attr_mngr
+
+    @staticmethod
+    def append_filters_to_telemetry_url(url: str, xdr_mode: bool, port_types: List[str]):
+        """
+        This function constructs and appends filter parameters to the given URL if certain conditions are met.
+
+        Parameters:
+            url (str): The base telemetry URL to which filters may be appended.
+            xdr_mode (bool): A flag indicating whether extended data record (XDR) mode is enabled.
+            port_types (List[str]): list of port type names used to generate filters.
+
+        Returns:
+            str: The telemetry URL with appended filter parameters if applicable, or the original URL.
+        """
+        filters = []
+        if xdr_mode:
+            filters.append(prepare_port_type_http_telemetry_filter(port_types))
+        if filters:
+            filters_sign = '&' if '?' in url else '?'
+            return f'{url}{filters_sign}{"&".join(filters)}'
+        return url
+
+    def get_metrics(self, _host, _port, _url, msg_tag):
+        _host = f'[{_host}]' if Utils.is_ipv6_address(_host) else _host
+        url = f'http://{_host}:{_port}/{_url}'
+        logging.info('Send UFM Telemetry Endpoint Request, Method: GET, URL: %s', url)
+        try:
+            response = requests.get(url)  # pylint: disable=missing-timeout
```

Comment:
add timeout for the get request (can be configurable)

Reply:
done